src/napari_matplotlib/scatter.py

```diff
@@ -1,39 +1,239 @@
+from typing import List, Tuple, Union
+
 import matplotlib.colors as mcolor
 import napari
+import numpy as np
+from magicgui import magicgui
 
 from .base import NapariMPLWidget
 
-__all__ = ["ScatterWidget"]
+__all__ = ["ScatterWidget", "FeaturesScatterWidget"]
 
 
-class ScatterWidget(NapariMPLWidget):
-    """
-    Widget to display scatter plot of two similarly shaped layers.
+class ScatterBaseWidget(NapariMPLWidget):
+    def __init__(
+        self,
+        napari_viewer: napari.viewer.Viewer,
+        marker_alpha: float = 0.5,
+        histogram_for_large_data: bool = True,
```

Comment:
I think I'd rather keep these as private class attributes for now, to avoid creating more public API. I'll open a separate issue for tracking the general problem of deciding when to switch between scatter/hist2d, and whether that should be user configurable.

Reply:
That makes sense to me. I will make them private properties.

Reply:
Done!